sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/SecretClientLiveTests.cs

```diff
@@ -0,0 +1,265 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Azure.Security.KeyVault.Secrets;
+using Azure.Core.Testing;
+
+namespace Azure.Security.KeyVault.Test
+{
+    public class SecretClientLiveTests : KeyVaultTestBase
+    {
+        private const int PagedSecretCount = 50;
+
+        public SecretClientLiveTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        [Test]
+        public async Task CrudWithExtendedProps()
+        {
+            string secretName = Recording.GenerateId();
+
+            Secret setResult = null;
+
+            try
+            {
+                var exp = new DateTimeOffset(new DateTime(637027248124480000, DateTimeKind.Utc));
+                var nbf = exp.AddDays(-30);
+
+                var secret = new Secret(secretName, "CrudWithExtendedPropsValue1")
+                {
+                    ContentType = "password",
+                    NotBefore = nbf,
+                    Expires = exp,
+                    Tags =
+                    {
+                        {"tag1", "value1"},
+                        {"tag2", "value2"}
+                    },
+                };
+
+                setResult = await Client.SetAsync(secret);
```

Comment:
Although we use `SetAsync` in every method, we don't have a specific test for it (i.e one that takes the full secret and one that takes name and value). 
Worth adding it separate in case we ever break it, it will be easy to know why all the other tests fail

Reply:
Added